src/worker.cc

```diff
@@ -17,16 +17,32 @@ inline WorkerServiceImpl::WorkerServiceImpl(const std::string& worker_address)
 }
 
 Status WorkerServiceImpl::ExecuteTask(ServerContext* context, const ExecuteTaskRequest* request, ExecuteTaskReply* reply) {
-  task_ = std::unique_ptr<Task>(new Task(request->task())); // Copy task
   RAY_LOG(RAY_INFO, "invoked task " << request->task().name());
-  WorkerMessage message = { &task_ };
-  RAY_CHECK(send_queue_.send(&message), "error sending over IPC");
+  std::unique_ptr<WorkerMessage> message(new WorkerMessage());
+  message->task = request->task();
+  {
+    WorkerMessage* message_ptr = message.get();
+    RAY_CHECK(send_queue_.send(&message_ptr), "error sending over IPC");
+  }
+  message.release();
+  return Status::OK;
+}
+
+Status WorkerServiceImpl::ImportFunction(ServerContext* context, const ImportFunctionRequest* request, ImportFunctionReply* reply) {
+  std::unique_ptr<WorkerMessage> message(new WorkerMessage());
+  message->function = request->function().implementation();
+  RAY_LOG(RAY_INFO, "importing function");
+  {
+    WorkerMessage* message_ptr = message.get();
+    RAY_CHECK(send_queue_.send(&message_ptr), "error sending over IPC");
+  }
+  message.release();
```

Comment:
It's doing the opposite. `release` releases ownership of the memory, not the memory itself.